apis/datadoghq/v1alpha1/datadogagentprofile_validation.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package v1alpha1
+
+import (
+	"fmt"
+)
+
+// IsValidDatadogAgentProfile is used to check if a DatadogAgentProfileSpec is valid
+func IsValidDatadogAgentProfile(spec *DatadogAgentProfileSpec) error {
```

Comment:
Nit: I think the name is a bit confusing. If it starts with `Is`, I'd expect the function to return a boolean instead of an error.
I think we should rename it to `ValidateDatadogAgentProfileSpec` or similar.